### ISSUE TITLE
[Feature] Implement importance weighted moment matching

### ISSUE THREAD

**Issue by n-kall** [arviz-devs/arviz-stats#63]
## Description

Importance weighted moment matching (IWMM) is an implicitly adaptive importance sampling method that improves proposal distributions in importance sampling by performing affine transformations on the proposal draws. This can be used in e.g. loo-cv or power-scaling sensitivity analysis, when Pareto-smoothed importance sampling is not stable.

The paper describing the method is available here: https://link.springer.com/article/10.1007/s11222-020-09982-2 and a reference method is implemented in R: https://github.com/topipa/iwmm
The loo R package (https://github.com/stan-dev/loo) also has an implementation.

## Thoughts on implementation

In order to run IWMM, I think the following are required:
- unconstrained parameter draws
- method to compute the posterior density for specified parameter values
- method to transform unconstrained draws to constrained draws

I'm happy to help with this, as I have been working with the R code and translating it should not be too hard, however I'm unsure of the best approach to get access to the above requirements (as they aren't usually in an InferenceData object)

**Comment by aloctavodia** [comment]
Yes, this will be implemented in this repo (I just transfered the issue)

**Comment by mlpotter** [comment]
@aloctavodia that is great to hear. Is there an expected timeline / release time for this functionality?

**Comment by aloctavodia** [comment]
I hope within the next ~4 months.

**Comment by jordandeklerk** [comment]
Working on this now. Hoping to have a solid initial implementation of this by end of this month!
